juju/juju
# refactor(apiserver): cutover apiserver get charm

Depends on: https://github.com/juju/juju/pull/18499

Cutover apiserver get charm object handler to be DQLite service backed.
This reduced conplexity significantly

To make this handler more testable, I have wrapped our application
service getter in an interface.

Move the existing tests for this handler to a legacy test file, and
start moving tests over. The old tests used the deprecated factory
methods.

## QA steps

```
$ juju bootstrap lxd lxd
$ juju add-model m
$ juju deploy ubuntu
$ juju add-unit ubuntu
```
Both units should become active-idle
